framework_tvb/tvb/adapters/visualizers/pse_isocline.py

```diff
@@ -129,7 +129,8 @@ def get_input_name():
 
     @staticmethod
     def get_filters():
-        return FilterChain(fields=[FilterChain.datatype + ".no_of_ranges"], operations=["=="], values=[2])
+        return FilterChain(fields=[FilterChain.datatype + ".no_of_ranges", FilterChain.datatype + ".display_isocline"],
+                           operations=["==", "=="], values=[2, 1])
```

Comment:
It seems to me like a double negation here... in this situation in fact we do not have a 2D range, only 1D. 
I would not add the extra parameter.
How about checking the ranges before submit to execution? When one of the two (range-param-1 or range-param-2) range choices returns a single value, cut it as a range, and convert it into a static value. I am not convinced when this should happen. It looks like a validation process, so maybe it belongs in the controllers layer?

Reply:
Why would it be a double negation? There are cases when the first condition can be true, but the second one false.

I am not sure I understand this:  "cut it as a range, and convert it into a static value". Does that mean I should set it to one value and let the no_of_ranges be 1? Otherwise without my new condition the Isocline Viewer would still show up.

Reply:
I agree it would be better to have a stronger validation on the ranges (and ensure that no single value range is generated instead of keeping extra fields in DB). Currently, we just do the `fill_from_post` which prepares the `RangeParameter` object and passes it along to the execution, without checking the actual values.
Maybe there we could also check in there how many values would each range have and in case there is only one, don't create the `RangeParameter` object, but directly set the single value on the session simulator? I think we have enough information to do that.

Reply:
> Why would it be a double negation? There are cases when the first condition can be true, but the second one false.
> 
> I am not sure I understand this: "cut it as a range, and convert it into a static value". Does that mean I should set it to one value and let the no_of_ranges be 1? Otherwise without my new condition the Isocline Viewer would still show up.

Yes, I think that is what Lia meant.